sfge/src/main/java/com/salesforce/rules/multiplemassschemalookup/MultipleMassSchemaLookupVisitor.java

```diff
@@ -0,0 +1,74 @@
+package com.salesforce.rules.multiplemassschemalookup;
+
+import com.salesforce.graph.symbols.SymbolProvider;
+import com.salesforce.graph.vertex.BaseSFVertex;
+import com.salesforce.graph.vertex.MethodCallExpressionVertex;
+import com.salesforce.graph.vertex.SFVertex;
+import com.salesforce.rules.ops.LoopDetectionVisitor;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Visitor detects when more than one invocation of Schema.getGlobalDescribe() is made in a path.
+ */
+class MultipleMassSchemaLookupVisitor extends LoopDetectionVisitor {
+    /** Represents the path entry point that this visitor is walking */
+    private final SFVertex sourceVertex;
+
+    /** Represents the mass schema lookup vertex that we're looking earlier calls for. */
+    private final MethodCallExpressionVertex sinkVertex;
+
+    /** Collects violation information */
+    private final HashSet<MultipleMassSchemaLookupInfo> violations;
+
+    /** Indicates if the sink vertex has been visited or not. */
+    private boolean isSinkVisited = false;
+
+    MultipleMassSchemaLookupVisitor(SFVertex sourceVertex, MethodCallExpressionVertex sinkVertex) {
+        this.sourceVertex = sourceVertex;
+        this.sinkVertex = sinkVertex;
+        this.violations = new HashSet<>();
+    }
+
+    @Override
+    protected void execAfterLoopVertexVisit(BaseSFVertex vertex, SymbolProvider symbols) {
+        if (shouldContinue()) {
```

Comment:
I'm not quite sure I understand what's going on here. This looks like it would throw a violation for:
```
for (int i = 0; i < 5; i++) {
//some code
}}
getGlobalDescribe()
```
How does it avoid doing that? I don't understand how it knows when it exits the loop.

Reply:
Ahh! I'm just dumb. You're spot on. Let me rework this.